### PR TITLE
add initial pass at arg_constraints and arg validation.

### LINKED ISSUE #120: pytorch-struct shows a warning for torch versions >= 1.8.0
When using `pytorch-struct` while using `torch>=1.8.0`, the following warning appears:
```
In [1]: import torch
   ...: from torch_struct import DependencyCRF, LinearChainCRF

In [2]: vals = torch.zeros(2, 10, 10) + 1e-5
   ...: vals[:, :5, :5] = torch.rand(5)
   ...: vals[:, 5:, 5:] = torch.rand(5)
   ...: dist = DependencyCRF(vals.log())
/Users/goncalocorreia/.pyenv/versions/3.8.1/envs/struct/lib/python3.8/site-packages/torch/distributions/distribution.py:44: UserWarning: <class 'torch_struct.distributions.DependencyCRF'> does not define `arg_constraints`. Please set `arg_constraints = {}` or initialize the distribution with `validate_args=False` to turn off validation.
  warnings.warn(f'{self.__class__} does not define `arg_constraints`. ' +
```

### PR DESCRIPTION
**Pull request by erip**
Fixes #120.

### CODE REVIEW

**Comment by erip** [comment]
I'm a bit sketchy on some of the parameters which get overloaded as tuples (i.e., `log_potentials` in `SentCFG`), so review on this seems particularly relevant. Everything else is pretty vanilla.

**Comment by srush** [comment]
SentCFG is a tuple of reals (triple). Does that work with constraints?

**Review by srush — APPROVED** [review]

**Comment by erip** [comment]
I guess that's the question. It seems like typically the expectation is parameters are not composite so you can have a clean mapping of parameter -> constraint. Since log_potentials is still technically a real (just in higher dim space) it _should_ be ok... I can look into adding a test to see if we can validate.

**Comment by erip** [comment]
As an aside, is there a reason CI didn't run on this PR? The GH Actions config looks OK so I'm not sure why I can't see the results of the tests.

**Comment by srush** [comment]
Kicked off the CI.
